src/Entity/AdherentFormation/Formation.php

```diff
@@ -335,4 +347,36 @@ public function incrementPrintCount(): void
     {
         ++$this->printCount;
     }
+
+    /**
+     * @return Collection|Adherent[]
+     */
+    public function getPrintByAdherents(): Collection
+    {
+        return $this->printByAdherents;
+    }
+
+    public function setPrintByAdherents(Collection $printByAdherents): void
+    {
+        $this->printByAdherents = $printByAdherents;
+    }
+
+    public function addPrintByAdherent(Adherent $adherent): void
+    {
+        if (!$this->printByAdherents->contains($adherent)) {
+            $this->printByAdherents->add($adherent);
+        }
+    }
```

Comment:
we can increment the count on `addPrintByAdherent` method and return `true` or `false` for flushing
```suggestion
    public function addPrintByAdherent(Adherent $adherent): bool
    {
        if (!$this->printByAdherents->contains($adherent)) {
            $this->printByAdherents->add($adherent);
            $this->incrementPrintCount();
            
            return true;
        }
        
        return false;
    }
```